eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/Mutators.scala

```diff
@@ -0,0 +1,50 @@
+package fr.acinq.eclair.wire.internal.channel
+
+import fr.acinq.bitcoin.scalacompat.{ByteVector32, Transaction}
+import fr.acinq.eclair.channel.{ClosingFeerates, Commitments, RealScidStatus, ShortIds}
+import fr.acinq.eclair.wire.protocol._
+import fr.acinq.eclair.{Alias, BlockHeight, RealShortChannelId}
+import shapeless.{::, HNil}
+
+/**
+ * This class contains mutators for [[shapeless.HList]] that are meant to be stacked to account for incremental minor
+ * changes to classes.
+ *
+ * Usage:
+ * {{{
+ *   def codec: Codec[MyClass] = (
+ *   // original codec
+ *   ...
+ *   )
+ *   .map(mutator1)
+ *   .map(mutator2)
+ *   .map(mutator3)
+ *   .decodeOnly
+ *   .as[MyClass]
+ * }}}
+ */
+object Mutators {
```

Comment:
Should we put that in a versioned namespace (this one would go in the `version3` package since we introduced the need for it in `version3` and won't need it for future versions)?

Reply:
Maybe, although we sort of adopted the opposite approach for `ChannelTypes0`.

Reply:
On the contrary, we put `ChannelTypes0` in the `version1` package? But it doesn't matter right now, we can move those mutators to a versioned package when we introduce new ones that are for a different codecs version.